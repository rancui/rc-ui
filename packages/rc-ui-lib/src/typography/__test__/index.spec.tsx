import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Typography from '..';

describe('Typography', () => {
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('render correctly', async () => {
    const { container } = render(<Typography.Text>这是一条文本</Typography.Text>);
    expect(container).toMatchSnapshot();
  });

  it('render correctly when using type prop', async () => {
    const { container } = render(<Typography.Text type="danger">这是一条文本</Typography.Text>);
    expect(container).toMatchSnapshot();
  });

  it('render correctly when using underline prop', async () => {
    const { container } = render(<Typography.Text underline>这是一条文本</Typography.Text>);
    expect(container).toMatchSnapshot();
  });

  it('render correctly when using ellipsis prop', async () => {
    const { container } = render(
      <Typography.Text ellipsis>
        In the process of internal desktop applications development, many different design specs and
        implementations would be involved
      </Typography.Text>,
    );
    expect(container).toMatchSnapshot();
  });

  it('render correctly when ellipsis has value', async () => {
    const { container } = render(
      <Typography.Text ellipsis={2}>
        {' '}
        In the process of internal desktop applications development, many different design specs and
        implementations would be involved
      </Typography.Text>,
    );
    expect(container).toMatchSnapshot();
  });

  it('render correctly when using level prop', async () => {
    const { container } = render(<Typography.Title level={1}>一级测试标题</Typography.Title>);
    expect(container).toMatchSnapshot();
  });
});
