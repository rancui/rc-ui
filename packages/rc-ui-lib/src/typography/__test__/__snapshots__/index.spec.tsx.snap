// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography render correctly 1`] = `
<div>
  <div
    class="rc-typography rc-typography--md rc-typography--text"
  >
    这是一条文本
  </div>
</div>
`;

exports[`Typography render correctly when ellipsis has value 1`] = `
<div>
  <div
    class="rc-typography rc-typography--md rc-typography--text rc-multi-ellipsis--l2"
  >
     
    In the process of internal desktop applications development, many different design specs and implementations would be involved
  </div>
</div>
`;

exports[`Typography render correctly when using ellipsis prop 1`] = `
<div>
  <div
    class="rc-typography rc-typography--md rc-typography--text rc-ellipsis"
  >
    In the process of internal desktop applications development, many different design specs and implementations would be involved
  </div>
</div>
`;

exports[`Typography render correctly when using level prop 1`] = `
<div>
  <div
    class="rc-typography rc-typography--md rc-typography--title rc-typography--l1"
  >
    一级测试标题
  </div>
</div>
`;

exports[`Typography render correctly when using type prop 1`] = `
<div>
  <div
    class="rc-typography rc-typography--danger rc-typography--md rc-typography--text"
  >
    这是一条文本
  </div>
</div>
`;

exports[`Typography render correctly when using underline prop 1`] = `
<div>
  <div
    class="rc-typography rc-typography--md rc-typography--text rc-typography--underline"
  >
    这是一条文本
  </div>
</div>
`;
